### PR TITLE
Fixing relative file path converter that was url encoding the path

### PR DESCRIPTION
**Pull request by distantcam**
OctopusDeploy/Issues#5978

### CODE REVIEW

**Comment by distantcam** [comment]
[S3 step doesn't handle spaces in filenames correctly](https://trello.com/c/kjIb8d1Q/2445-s3-step-doesnt-handle-spaces-in-filenames-correctly)

**Comment by gupta-kartik** [comment]
LGTM as it's quite straightforward but (being new to the world of .NET) I would get a second pair of  👀 on it from @matt-richardson

File: source/Calamari.Tests/Fixtures/Util/StringExtensionsFixture.cs [hunk 1]
```diff
@@ -27,5 +28,19 @@ public void NullValueShouldThrowException()
             Action action = () => "foo".ContainsIgnoreCase(null);
             action.Should().Throw<ArgumentNullException>();
         }
+
+        // This method has some weird legacy issues which we now rely on
+        [TestCase(@"C:\Path\To\File1.txt", @"C:\Path\", @"To/File1.txt")]
+        [TestCase(@"C:\Path\To\File2.txt", @"C:\Path", @"To/File2.txt")]
+        [TestCase(@"C:\Path\To\File3 With Spaces.txt", @"C:\Path", @"To/File3 With Spaces.txt")]
+        //[TestCase(@"C:/Path/To/File4.txt", @"C:/Path/", @"To/File4.txt")]
```
Comment (matt-richardson) [inline source/Calamari.Tests/Fixtures/Util/StringExtensionsFixture.cs]: Whats the go with this test case? Can we uncomment or delete?

Comment (gupta-kartik) [reply source/Calamari.Tests/Fixtures/Util/StringExtensionsFixture.cs]: I was a bit confused here with the `/` `\` changing in assertions in the tests 1-3. Are we doing this to check that both types of slashes have the same result?

Comment (distantcam) [reply source/Calamari.Tests/Fixtures/Util/StringExtensionsFixture.cs]: It's a weird case which produces `../To/File4.txt` which is an unexpected output. I'd prefer to delete.

This method returns some unexpected results when it comes to non-windows style paths. I'm uncomfortable changing the behaviour given we rely on it elsewhere.

Comment (distantcam) [reply source/Calamari.Tests/Fixtures/Util/StringExtensionsFixture.cs]: I've removed that comment

**Pull request merged**